wallet/how-to/use-non-evm-networks/starknet/manage-starknet-networks.md

```diff
@@ -0,0 +1,140 @@
+---
+description: Manage Starknet networks in MetaMask.
+sidebar_position: 3
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+# Manage Starknet networks
+
+You can detect a user's Starknet network and prompt them to switch Starknet networks in MetaMask,
+using the
+[`get-starknet`](https://github.com/starknet-io/get-starknet) library or the
+[`wallet_invokeSnap`](/snaps/reference/wallet-api-for-snaps/#wallet_invokesnap) JSON-RPC method.
+
+## Prerequisites
+
+[Connect to Starknet](connect-to-starknet.md) from your dapp.
+
+## Detect a user's network
+
+Detect the Starknet network a user is currently connected to using the following:
+
+<Tabs>
+  <TabItem value="get-starknet" default>
+
+  ```javascript
+  const checkCurrentNetwork = () => {
+    const starknet = getStarknet();
+    const currentNetwork = starknet.provider.rpcUrl;
+    console.log("Currently connected to:", currentNetwork);
+    return currentNetwork;
+  };
+  ```
+  
+  </TabItem>
+  <TabItem value="wallet_invokeSnap">
+
+  ```javascript
+  const checkCurrentNetwork = async () => {
+    if (typeof window.ethereum !== "undefined" && window.ethereum.isMetaMask) {
+      try {
+        const response = await window.ethereum.request({
+          method: "wallet_invokeSnap",
+          params: {
+            snapId: "npm:@consensys/starknet-snap",
+            request: {
+              method: "starknet_getChainId"
+            }
+          }
+        });
+        
+        let networkName;
+        switch (response) {
+          case "0x534e5f4d41494e":
+            networkName = "Mainnet";
+            break;
+          case "0x534e5f5345504f4c4941":
+            networkName = "Sepolia Testnet";
+            break;
+          default:
+            networkName = "Unknown Network";
+        }
+        
+        console.log("Currently connected to:", networkName);
+        return response;  // Returns the chain ID.
+      } catch (error) {
+        console.error("Error getting current Starknet network:", error);
+        throw error;
```

Comment:
```suggestion
  const checkCurrentNetwork = async () => {
    const provider = await getEip6963Provider()

    if (provider) {
      try {
        const response = await provider.request({
          method: "wallet_invokeSnap",
          params: {
            snapId: "npm:@consensys/starknet-snap",
            request: {
              method: "starkNet_getCurrentNetwork"
            }
          }
        });
        
        console.log("Currently connected to:", response.name);
        return response.chainId;  // Returns the chain ID.
      } catch (error) {
        console.error("Error getting current Starknet network:", error);
        throw error;
```

we should always guide user to use eip6963 to  get the provider, instead of using window.ethereum
hence i create a mock implementation `const provider = await getEip6963Provider()`

Reply:
This is a bit confusing @stanleyyconsensys. We can add a note to use the recommended EIP-6963, but for all the other methods we are using `window.ethereum`. For consistency could we provide both, or maybe I can include a note to flag this,